src/main/java/com/findmyflight/findmyflight/service/emailnotificationreceiver/EmailNotificationReceiverService.java

```diff
@@ -0,0 +1,50 @@
+package com.findmyflight.findmyflight.service.emailnotificationreceiver;
+
+import com.findmyflight.findmyflight.service.error.handler.AddressExistException;
+import com.findmyflight.findmyflight.service.error.handler.IdNotExistException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class EmailNotificationReceiverService {
+    private final EmailNotificationReceiverRepository repository;
+    private final EmailNotificationReceiverMapper mapper;
+
+    @Transactional
+    public void createEmailNotificationReceiver(EmailNotificationReceiverDto emailNotificationReceiverDto) {
+        if (repository.existsByAddress(emailNotificationReceiverDto.address())){
+            throw new AddressExistException();
+        }else {
+            var emailNotificationReceiver = mapper.map(emailNotificationReceiverDto);
+            repository.save(emailNotificationReceiver);
+        }
+    }
+    @Transactional(readOnly = true)
+    public List<EmailNotificationReceiver> getAllEmailNotificationReceivers(){
+        return repository.findAll();
+    }
+    @Transactional
+    public void editEmailNotificationReceiver(Long id, EmailNotificationReceiverDto emailNotificationReceiverDto){
+        if(repository.existsById(id)){
+            var emailNotificationReceiver = repository.findById(id).get();
+            mapper.updateEmailNotificationReceiverFromDto(emailNotificationReceiverDto,emailNotificationReceiver);
+            repository.save(emailNotificationReceiver);
+        }
+        else {
+            throw new IdNotExistException();
+        }
+    }
+    @Transactional
+    public void deleteEmailNotificationReceiver(Long id) {
+        if (repository.existsById(id)) {
```

Comment:
Here, the same situation - 1 too many DB calls :) You can just write
```
repository.deleteById(id);
```
and don't check the existence of the entity.

That's because the REST API doesn't need to inform the client that there's no ID. The contract is that DELETE method should just make sure that the resource is deleted - so no harm if it wasn't created in the first place. That makes the DELETE method _**idempotent**_, which is very important for the REST API (read more [here](https://www.restapitutorial.com/lessons/idempotency.html) - the video is a bit over the top, but actually explains the subject nicely).